src/Avalonia.Controls/TreeViewItem.cs

```diff
@@ -65,6 +75,34 @@ public bool IsSelected
             set { SetValue(IsSelectedProperty, value); }
         }
 
+        /// <summary>
+        /// Gets or sets the depth of the item.
+        /// </summary>
+        public int Depth
+        {
+            get { return this.GetDepth(this); }
+        }
+
+        private int GetDepth(TreeViewItem item)
+        {
+            TreeViewItem parent;
+            while ((parent = GetParent(item)) != null)
+            {
+                return GetDepth(parent) + 1;
+            }
+            return 0;
+        }
+
+        private static TreeViewItem GetParent(TreeViewItem item)
+        {
+            var parent = item.InheritanceParent;
```

Comment:
One question: why use `InheritanceParent` here rather than just `Parent`?

Reply:
Changed that earlier... 👍